data_structures/linked_list/circular_doubly_linked_list.c

```diff
@@ -0,0 +1,307 @@
+ * the first node points to the last node via the previous pointer.
+ *
+ * In this implementation, the menu-driven code is written along with functions
+ * to insert at the head, insert at the last index, delete the first node,
+ * delete the last node, and display list functions are coded.
+ *
+ * @author [Sahil Kandhare](https://github.com/SahilK-027)
+ *
+ */
+
+#include <inttypes.h>  /// to provide a set of integer types with universally consistent definitions that are operating system-independent
+#include <stdio.h>     /// for IO operations
+#include <stdlib.h>  /// for including functions involving memory allocation such as `malloc`
+
+/**
+ * @brief Circular Doubly linked list struct
+ *
+ */
+typedef struct node
+{
+    struct node *prev, *next;  ///< List pointers
+    uint64_t value;            ///< Data stored on each node
+} ListNode;
+
+/**
+ * @brief Create a list node
+ *
+ * @param data
+ * @return ListNode*  newly created list node
+ */
+ListNode *create_list(uint64_t data)
+{
+    ListNode *new_list = (ListNode *)malloc(sizeof(ListNode));
+    new_list->value = data;
+    new_list->next = new_list;
+    new_list->prev = new_list;
+    return new_list;
+}
+
+/**
+ * @brief Insert the node at start of list
+ *
+ * @param head   starting pointer of list
+ * @param data   The data which we want to insert into list
+ * @return ListNode*
+ */
+ListNode *insert_at_head(ListNode *head, uint64_t data)
+{
+    if (head == NULL)
+    {
+        head = create_list(data);
+        return head;
+    }
+    else
+    {
+        ListNode *temp;
+        ListNode *new_node = create_list(data);
+        temp = head->prev;
+        new_node->next = head;
+        head->prev = new_node;
+        new_node->prev = temp;
+        temp->next = new_node;
+        head = new_node;
+        return head;
+    }
+}
+
+/**
+ * @brief Insert the node at end of list
+ *
+ * @param head      starting pointer of list
+ * @param data      The data which we want to insert into list
+ * @return ListNode*
+ */
+ListNode *insert_at_last(ListNode *head, uint64_t data)
+{
+    if (head == NULL)
+    {
+        head = create_list(data);
+        return head;
+    }
+    else
+    {
+        ListNode *temp1, *temp2;
+        ListNode *new_node = create_list(data);
+        temp1 = head;
+        temp2 = head->prev;
+        new_node->prev = temp2;
+        new_node->next = temp1;
+        temp1->prev = new_node;
+        temp2->next = new_node;
+        return head;
+    }
+}
+
+/**
+ * @brief  Function for deletion of the first node in list
+ *
+ * @param head      starting pointer of list
+ * @return ListNode*
+ */
+ListNode *delete_first_node(ListNode *head)
+{
+    if (head == NULL)
+    {
+        printf("list is Empty\n");
+        return head;
+    }
+    ListNode *temp1, *temp2;
+    temp1 = head;
+    temp2 = temp1->prev;
+    if (temp1 == temp2)
+    {
+        free(temp2);
+        head = NULL;
+        printf("\nNode succesfully deleted\n\n");
```

Comment:
Well, apparently every `successfully` in your code have this typo:thinking:
```suggestion
        printf("\nNode successfully deleted\n\n");
```

Reply:
Actually, I wrote that statement once and then copy-pasted it throughout the code. 😅